crates/transaction-pool/src/pool/mod.rs

```diff
@@ -745,6 +754,14 @@ struct PendingTransactionListener {
     kind: PendingTransactionListenerKind,
 }
 
+/// An active listener for new pending transactions.
+#[derive(Debug)]
+struct TransactionListener<T: PoolTransaction> {
+    sender: mpsc::Sender<NewTransactionEvent<T>>,
+    /// Whether to include transactions that should not be propagated over the network.
+    kind: PendingTransactionListenerKind,
```

Comment:
Not that happy about reusing `PendingTransactionListenerKind` here. If this change is correct, how would you design this?

Reply:
probably just rename it to `TransactionListenerKind` and replace the references in the docs for "pending" with something like "in this pool", wdyt @mattsse?

Reply:
yeah I like that, `TransactionListenerKind` is more appropriate